### ISSUE TITLE
CdmBook Toolchain: Allow for creation of packages when original OBJ is not available

### ISSUE THREAD

**Issue by MarcusBarnes** [MarcusBarnes/mik#206]
In certain circumstances, original source OBJ files (e.g., original TIFF files) may be (temporarily) unavailable.  Allow creation of book packages using data from Cdm only.  This will require using either JP2 or JPEG from Cdm as the required OBJ data stream for the packages.

N.B. Once the packages have been loaded into Islandora, the OBJ data streams may be updated with the original source OBJ files using a tool such as https://github.com/mjordan/islandora_datastream_crud

**Comment by MarcusBarnes** [comment]
Some initial work on this issue have been pushed to https://github.com/MarcusBarnes/mik/tree/issue-206.

Two more things need to be completed before closing this issue and making a pull-request:

1.  Assign OBJ DSID to a copy of JPEG or JP2 files (depending on which is available).

2.  If the package for the book has already been written to the output directory, make a duplicate.

**Comment by MarcusBarnes** [comment]
If no original source OBJ are available for use with the OBJ datastream, make a copy of JP2 as OBJ.jp2 or if that is not available make a copy of the JPEG as OBJ.jpeg:  https://github.com/MarcusBarnes/mik/commit/1eed4a6207efe0888f95d54d12b010f0953df62c

I'd like to add one additional setting to the writer section, something like skip_obj in cases where we are not creating import packages, but for example, just exporting specific datastreams (in a structured manner) for testing or other purposes (e.g., MODS only).

**Comment by MarcusBarnes** [comment]
In commit https://github.com/MarcusBarnes/mik/commit/53a8d6c44c148a06caff1bbcabce5ccb3cb2c2f5  I've added the ability to skip OBJ datastream creation by adding skip_obj = true to the [WRITER] section of the configuration file.  This may be desirable should you wish to create test runs on specific datastreams other than the OBJ datastream.

@mjordan  Would you please give https://github.com/MarcusBarnes/mik/tree/issue-206 a test and create a pull-request if everything works as expected?  Thank you.

**Comment by MarcusBarnes** [comment]
Merged into master with https://github.com/MarcusBarnes/mik/commit/2848125e6b17bdf4e3248cdb585263e36e0c1d2b.
